core/coreapi/interface/dht.go

```diff
@@ -0,0 +1,24 @@
+package iface
+
+import (
+	"context"
+
+	options "github.com/ipfs/go-ipfs/core/coreapi/interface/options"
+
+	ma "gx/ipfs/QmWWQ2Txc2c6tqjsBpzg5Ar652cHPGNsQQp2SejkNmkUMb/go-multiaddr"
+	peer "gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
+)
+
+// DhtAPI specifies the interface to the DHT
+type DhtAPI interface {
```

Comment:
I'd rather use the interface defined in https://github.com/libp2p/go-libp2p-routing/ (although we'd probably want to change it to return channels). Alternatively, we could do this for now with a big "we're going to change this" todo.